scalafix-reflect/src/main/scala/scalafix/internal/reflect/RuleInstrumentation.scala

```diff
@@ -15,26 +15,26 @@ object RuleInstrumentation {
       def unapply(templ: Template): Boolean = templ match {
 
         // v0
-        case Template(_, init"Rewrite" :: _, _, _) => true
-        case Template(_, init"Rule($_)" :: _, _, _) => true
-        case Template(_, init"SemanticRewrite($_)" :: _, _, _) => true
-        case Template(_, init"SemanticRule($_, $_)" :: _, _, _) => true
+        case Template(_, Init(Type.Name("Rewrite"), _, _) :: _, _, _) => true
+        case Template(_, Init(Type.Name("Rule"), _, List(List(_))) :: _, _, _) => true
+        case Template(_, Init(Type.Name("SemanticRewrite"), _, List(List(_))) :: _, _, _) => true
+        case Template(_, Init(Type.Name("SemanticRule"), _, List(List(_, _))) :: _, _, _) => true
```

Comment:
@tgodzik can you please confirm it's correct. We are rewriting quasiqotes to be able to compile in scala 3. thank you